### PR TITLE
[editor] hello world local web app

### PR DESCRIPTION
**Pull request by jonathanlastmileai**
[editor] hello world local web app

- flask server / static react front-end
- add python run script (this can be aliased as `aiconfig`)
- use node postbuild script to copy the react files to web static folder
  so the web app runs as expected (see https://nitratine.net/blog/post/how-to-serve-a-react-app-from-a-python-server/#goal)
- package run script and server with python-aiconfig package
- readme says how to modify, build, run and use
- tl;dr
  - pip install python-aiconfig
  - set aiconfig alias
  - run server with script
  - open localhost

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/530).
* __->__ #530
* #513[editor] hello world editor server

should be pretty self-explanatory

### CODE REVIEW

File: python/pyproject.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 requires = ["setuptools", "wheel"]
 
 [project]
-name = "python-aiconfig"
+name = "python-aiconfig-test"
```
Comment (saqadri) [inline python/pyproject.toml:5]: Undo

File: python/src/aiconfig/scripts/aiconfig.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import asyncio
```
Comment (saqadri) [inline python/src/aiconfig/scripts/aiconfig.py:1]: Name of this file is misleading. How about `aiconfig_cli.py`?

File: python/src/editor/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+# Install:
+
+Install python-aiconfig from pip, then set `aiconfig` alias (in shell, .bashrc, .zshrc, etc.)
+
+One-liner:
+`pip install python-aiconfig; alias aiconfig="python -m 'aiconfig.scripts.aiconfig'"`
+
+## Usage
+
+`aiconfig edit  --aiconfig-path=/my/path'`
```
Comment (saqadri) [inline python/src/editor/README.md:12]: just trying to understand if this is the recommended way to run this, or if we'll package this into an actual CLI utility?

File: python/src/editor/client/.gitignore [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
Comment (saqadri) [inline python/src/editor/client/.gitignore:1]: Is this needed?

File: python/src/editor/client/package-lock.json
Comment (saqadri) [inline python/src/editor/client/package-lock.json:1]: Delete. We should use yarn instead (so there should be a yarn.lock)

**Review by saqadri — APPROVED** [review]
LGTM, just some nits

File: python/src/editor/README.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+[INFO] 2023-12-18 23:54:14,379 server.py:33: Editor server running on http://localhost:8888
+```
+
+Go to url in browser to use app.
+
+## Dev
+
+1. Edit
+   Server code: `python/src/editor/server/server.py`
+   Client code: `aiconfig/python/src/editor/client/src`
```
Comment (rholinshead) [inline python/src/editor/README.md:28]: Can we move these under /cli instead?

File: python/src/editor/README.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+2. Build react app: `npm run build`
```
Comment (rholinshead) [inline python/src/editor/README.md:30]:
```suggestion
2. Build react app: `yarn build`
```

File: python/src/editor/README.md (hunk 3 above)
Comment (jonathanlastmileai) [reply python/src/editor/README.md:12]: I think we can wrap this in a shell script and distribute it with pip. I don't know how cross-platform that would be compared to running python directly this way.

File: python/src/editor/client/.gitignore (hunk 4 above)
Comment (jonathanlastmileai) [reply python/src/editor/client/.gitignore:1]: nope will remove

File: python/src/editor/README.md (hunk 3 above)
Comment (jonathanlastmileai) [reply python/src/editor/README.md:12]: for ref: https://dzone.com/articles/executable-package-pip-install

Comment (jonathanlastmileai) [reply python/src/editor/README.md]: https://docs.google.com/document/d/1jUsV3Khu_Gs31dIBxh1n2DnF_AbnRPiN9etZpLe1dk8/edit?usp=sharing

File: python/src/editor/README.md (hunk 5 above)
Comment (jonathanlastmileai) [reply python/src/editor/README.md]: This might break the packaging so I'm going to suggest we punt on this. Leaving it here for now won't affect UX. 

https://docs.google.com/document/d/1jUsV3Khu_Gs31dIBxh1n2DnF_AbnRPiN9etZpLe1dk8/edit?usp=sharing

File: python/src/editor/README.md (hunk 6 above)
Comment (jonathanlastmileai) [reply python/src/editor/README.md]: punt https://docs.google.com/document/d/1jUsV3Khu_Gs31dIBxh1n2DnF_AbnRPiN9etZpLe1dk8/edit?usp=sharing
